dessia_common/datatools/modeling.py

```diff
@@ -0,0 +1,734 @@
+"""
+Tools and base classes for machine learning methods.
+
+"""
+from typing import List, Dict, Any
+
+import numpy as npy
+from sklearn import preprocessing, linear_model, ensemble, tree, svm, neural_network
+
+from dessia_common.core import DessiaObject
+
+
+# ======================================================================================================================
+#                                                     S C A L E R S
+# ======================================================================================================================
+class BaseScaler(DessiaObject):
+    _rebuild_attributes = []
+
+    def __init__(self, name: str = ''):
+        DessiaObject.__init__(self, name=name)
+
+    @classmethod
+    def _skl_class(cls):
+        raise NotImplementedError('Method _skl_class not implemented for BaseScaler. Please use children.')
+
+    def _call_skl_scaler(self):
+        return self._skl_class()()
+
+    def instantiate_skl(self):
+        scaler = self._call_skl_scaler()
+        for attr in self._rebuild_attributes:
+            setattr(scaler, attr, getattr(self, attr))
+        return scaler
+
+    @classmethod
+    def instantiate_dessia(cls, scaler, name: str = ''):
```

Comment:
If you wish to type scaler with the BaseScaler type, consider using the forward reference typing : 
`def method(cls, scaler : 'BaseScaler', name: str = ""):`
This works perfectly fine and is supported by dessia_common. I don't know if this is what you intend to do, though. 
Just a comment anyway, if the method is not meant to be ran on platform that's purely optional and for documentation purpose (for example, it would have help understand this bit of code faster :) )

Reply:
Unfortunatly, the scaler here is the scikit-learn object from `sklearn.preprocessing.StandardScaler`, which cannot be serialized.
Maybe I should write it ?
`def method(cls, scaler : preprocessing.StandardScaler, name: str = ""):`

And the same for models below ?

Reply:
I guess you could. Is this function meant to be called from the platform ? (and thus, be opened in a form) In this case, preprocessing.StandardScaler cannot work as you need typings supported by dessia_common. 
In the other case, its just for doc purpose and not mandatory at all. Actually sometimes I even advise against it, so I let you do whatever you prefer on this

Reply:
After some reflexion and a try it appears that the sklearn scaler expected here has not a predefined type. It can be StandardScaler, LinearScaler, 'HardName',...

So I decide not to set it, except if i can write this : 
`def instantiate_dessia(cls, scaler: preprocessing, name: str = '')`

which i guess is not possible since preprocessing is a module...

Are you ok ?